producer.go

```diff
@@ -1352,13 +1352,22 @@ func (self *TopicProducerMgr) PublishWithExtAndRetryBackground(topic string, bod
 }
 
 func (self *TopicProducerMgr) Publish(topic string, body []byte) error {
-	_, err := self.doCommandWithRetry(topic, nil, func(pid int) (*Command, error) {
+	respByte, err := self.doCommandWithRetry(topic, nil, func(pid int) (*Command, error) {
```

Comment:
The failed publish should return an error if resp is not OK. What is  the `respByte` while your error is nil and pub failed?

Reply:
I tested and found `string(respByte)` is `""` while error is nil and pub failed. The failed publish not always return an error.

Reply:
I can not reproduce this. Can you give some codes and logs to show how to reproduce it?